### ISSUE TITLE
Getting Error: read EALREADY when calling connect

### ISSUE THREAD

**Issue by unbornchikken** [mscdex/node-ftp#109]
I have an IIS FTPS server.

Connect options are:

```
{
  host: "server01",
  secure: true,
  user: "server01|user01",
  password: "password01"
}
```

I'm getting "Error: read EALREADY" exception when calling connect. It's originated from connection.js line 268, where node-ftp calls tls.connect.

The server is working I can connect with Total Commander without problems.

I'm using io.js 1.4.1.

**Comment by mscdex** [comment]
Can you try with io.js 1.3.0 or node 0.12?

**Comment by unbornchikken** [comment]
Thanks for looking at it. Unfortunattely I have to go, but tomorrow morning Im gonna try and feedback.

**Comment by unbornchikken** [comment]
My code is using ES6 features that Node.js 0.12 ain't support, so I did try it with io.js 1.3. This time I get:

```
Error: unable to verify the first certificate
    at Error (native)
    at TLSSocket.<anonymous> (_tls_wrap.js:896:36)
    at emitNone (events.js:67:13)
    at TLSSocket.emit (events.js:163:7)
    at TLSSocket._finishInit (_tls_wrap.js:438:8)
```

Please note if I go with

```
secure: false
```

node-ftp works with all versions of io.js.

**Comment by unbornchikken** [comment]
I went with NODE_TLS_REJECT_UNAUTHORIZED=0 and I've got:

```
Error: The supplied message is incomplete. The signature was not verified.
    at makeError (C:\GIT\SanomaMedia\node_modules\ftp\lib\connection.js:1071:13)
    at Parser.<anonymous> (C:\GIT\SanomaMedia\node_modules\ftp\lib\connection.js:113:25)
    at emitTwo (events.js:87:13)
    at Parser.emit (events.js:169:7)
    at Parser._write (C:\GIT\SanomaMedia\node_modules\ftp\lib\parser.js:58:10)
    at doWrite (_stream_writable.js:295:12)
    at writeOrBuffer (_stream_writable.js:282:5)
    at Parser.Writable.write (_stream_writable.js:210:11)
    at TLSSocket.ondata (C:\GIT\SanomaMedia\node_modules\ftp\lib\connection.js:281:20)
    at emitOne (events.js:77:13)
    at TLSSocket.emit (events.js:166:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at TLSSocket.Readable.push (_stream_readable.js:109:10)
    at TCP.onread (net.js:517:20)
```

**Comment by unbornchikken** [comment]
Ok I went with io.js 1.4.3 / NODE_TLS_REJECT_UNAUTHORIZED=0 and I've got the same error:

```
Error: The supplied message is incomplete. The signature was not verified.
    at makeError (C:\GIT\SanomaMedia\node_modules\ftp\lib\connection.js:1071:13)
    at Parser.<anonymous> (C:\GIT\SanomaMedia\node_modules\ftp\lib\connection.js:113:25)
    at emitTwo (events.js:87:13)
    at Parser.emit (events.js:169:7)
    at Parser._write (C:\GIT\SanomaMedia\node_modules\ftp\lib\parser.js:58:10)
    at doWrite (_stream_writable.js:295:12)
    at writeOrBuffer (_stream_writable.js:282:5)
    at Parser.Writable.write (_stream_writable.js:210:11)
    at TLSSocket.ondata (C:\GIT\SanomaMedia\node_modules\ftp\lib\connection.js:281:20)
    at emitOne (events.js:77:13)
    at TLSSocket.emit (events.js:166:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at TLSSocket.Readable.push (_stream_readable.js:109:10)
    at TCP.onread (net.js:517:20)
```

I did some research that suggests the above issue is originated from IIS, and there is a HF for it: http://support.microsoft.com/hotfix/kbhotfix?kbnum=2888853&kbln=en-US This HF is already installed in our ftps server, and as I said before Total Commander can connect / upload / download via ftps without issues. So this is not the case.

**Comment by unbornchikken** [comment]
I did some more research and it seems TC is working with my IIS FTPS server because it is not uses TLS 1.2. Better clients, like yours, are using TLS 1.2 protocol, and on this case I get "The supplied message is incomplete. The signature was not verified." from IIS. I've replaced this MS crap with FileZilla, and node-ftp based FTPS access gets started to work like charm.

So the result is:
- I'm using io.js 1.4.3 instead of 1.4.1
- I'm using FileZilla FTPS instead of MS IIS

Now the issue is gone.
